### PR TITLE
Unified kuadrant - Remove TLSPolicy and DNSPolicy

### LINKED ISSUE #767: Move DNSPolicy to Kuadrant Operator
**What**

We are bringing the policy controllers into a single controller. We want to move the logic, CRDS, integration and e2e test for DNSPolicy into the kuadrant operator.

**Done**

- [x] Controller , CRDS and Tests moved over to kuadrant operator
- [x] Relevant Docs moved to kuadrant operator
- [x] Any relevant links updated in the Docs navigation 

**Requires**

* https://github.com/Kuadrant/multicluster-gateway-controller/issues/779
* https://github.com/Kuadrant/kuadrant-dns-operator/issues/11

### PR DESCRIPTION
**Pull request by mikenairn**
close #767 #766 #784 

Removes all dns and tls related resources.

* Remove dnspolicy, tlspolicy, dnsrecord, dnshealth and managedzone api and reconcilers
* Remove all policy-controller related config and make targets
* Adds the kuadrant-operator as a dependency of the MGC operator installation and is installed on the OCM hub cluster during local-setup.

**Verification**

Tested with local-setup and e2e tests.

**Requires**

* https://github.com/Kuadrant/kuadrant-operator/pull/416

### CODE REVIEW

File: cmd/gateway_controller/main.go [hunk 1]
```diff
@@ -55,7 +54,6 @@ var (
 func init() {
 	utilruntime.Must(clientgoscheme.AddToScheme(scheme.Scheme))
 
-	utilruntime.Must(v1alpha1.AddToScheme(scheme.Scheme))
 	utilruntime.Must(certmanv1.AddToScheme(scheme.Scheme))
```
Comment (maleck13) [inline cmd/gateway_controller/main.go:57]: should this go away now too?

File: config/local-setup/issuer/issuer.yaml
Comment (maleck13) [inline config/local-setup/issuer/issuer.yaml:1]: Should this be just moved / removed?

File: cmd/gateway_controller/main.go (hunk 1 above)
Comment (mikenairn) [reply cmd/gateway_controller/main.go:57]: this can probably be removed

File: test/gateway_integration/suite_test.go [hunk 2]
```diff
@@ -93,8 +92,8 @@ var _ = BeforeSuite(func() {
 	Expect(err).NotTo(HaveOccurred())
 	Expect(cfg).NotTo(BeNil())
 
-	err = v1alpha1.AddToScheme(scheme.Scheme)
-	Expect(err).NotTo(HaveOccurred())
+	//err = v1alpha1.AddToScheme(scheme.Scheme)
```
Comment (maleck13) [inline test/gateway_integration/suite_test.go:95]: do we just want to remove if not needed?

File: config/local-setup/issuer/issuer.yaml
Comment (mikenairn) [reply config/local-setup/issuer/issuer.yaml:1]: Yeah, i updated local-setup to create this in the hub to keep the behaviour consistent.

I think it's fine to leave it, at least for now, it's only for local-setup and has been moved into the `config/local-setup` directory to reflect that.

File: test/gateway_integration/suite_test.go (hunk 2 above)
Comment (mikenairn) [reply test/gateway_integration/suite_test.go:95]: yep, can be removed

Comment (mikenairn) [reply test/gateway_integration/suite_test.go:95]: there is no api in this repo now, just left over from when i was working through it.

**Review by maleck13 — COMMENTED** [review]
changes all look fine. Just wondering if we need to review where stuff for cert-manager is used and if that should also be removed?

File: docs/how-to/multicluster-loadbalanced-dnspolicy.md [hunk 3]
```diff
@@ -111,7 +111,8 @@ The health check section is optional, the following fields are available:
 - `port`: The port to connect to
 - `protocol`: The protocol to use for this connection
 
-For more information about DNS Health Checks, see [this guide](../dnspolicy/dns-health-checks.md).
+[//]: # (ToDo mnairn)
```
Comment (maleck13) [inline docs/how-to/multicluster-loadbalanced-dnspolicy.md:114]: is this so we don't break docs when me merge?

File: test/gateway_integration/suite_test.go [hunk 4]
```diff
@@ -93,18 +91,15 @@ var _ = BeforeSuite(func() {
...
```
Comment (maleck13) [inline test/gateway_integration/suite_test.go:94]: should we remove this now?

File: docs/how-to/multicluster-loadbalanced-dnspolicy.md (hunk 3 above)
Comment (mikenairn) [reply docs/how-to/multicluster-loadbalanced-dnspolicy.md:114]: meh, i need to go back through the docs and link to to where they are now

Comment (mikenairn) [reply docs/how-to/multicluster-loadbalanced-dnspolicy.md]: Updated

**Review by maleck13 — APPROVED** [review]
went through multi-cluster walkthrough and quick start all worked as expected

**Comment by maleck13** [comment]
by passed e2e as this needs to be merged for them to start passing

**Pull request merged**